rlp/rlp.go

```diff
@@ -0,0 +1,43 @@
+package rlp
+
+import (
+	"encoding/binary"
+	"math"
+)
+
+type Item struct {
+	D []byte
+	L []*Item
+}
+
+func Encode(input *Item) []byte {
+	if input.D != nil {
+		if len(input.D) == 1 && input.D[0] < 128 {
+			return input.D
+		}
+		return append(encodeLength(input.D, 128), input.D...)
+	}
+
+	var out []byte
+	for i := range input.L {
+		out = append(out, Encode(input.L[i])...)
+	}
+	return append(encodeLength(out, 192), out...)
+}
+
+func encodeLength(input []byte, offset uint8) []byte {
+	switch l := uint64(len(input)); {
+	case l < 56:
+		header := uint64(offset) + l
+		return []byte{byte(header)} // (max(128,192) + 55) < 255
```

Comment:
Isn't it problematic if `offset` is greater than 192? Should you add a check here for that

Reply:
Good call. Let me think about how we can tighten this up.

Reply:
I did a pretty significant refactor in which i moved all the "magic" numbers to a const declaration. I hope this makes it easier to reason about.